### ISSUE TITLE
Implementation of Physics Project WFR functions

### ISSUE THREAD

**Issue by maxitg** [maxitg/SetReplace#462]
### The problem
There are many functions in the [Wolfram Function Repository](https://www.wolframcloud.com/obj/wolframphysics/Tools/guide-page) which would be handy to have in *SetReplace*.

It would be useful to make these functions more reliable, test them, possibly improve the design, and move them to *SetReplace*.

This issue tracks this effort. If you'd like to implement a particular function, create another issue just for that function, and link it next to one of the checkboxes below. ***It's best to discuss the API for the new functions before starting to implement them***. The design of some of the functions needs improvement. Not all WFR functions should map into functions. It's usually better to integrate a function into an existing function (such as a `WolframModel` property). Some of them don't currently work correctly, so be careful. If you know a useful WFR function that is not mentioned below, please add it to the list.

Note that [`MultiwaySystem`](https://resources.wolframcloud.com/FunctionRepository/resources/MultiwaySystem) is intentionally not included in the list below. We do want to implement it, but through the [Local Multiway System](https://github.com/maxitg/SetReplace/blob/master/Research/LocalMultiwaySystem/LocalMultiwaySystem.md), i.e., as a part of the [`WolframModel`](https://github.com/maxitg/SetReplace/blob/master/README.md#wolframmodel-and-wolframmodelevolutionobject) function. Similarly, functions like [`CausalInvariantQ`](https://resources.wolframcloud.com/FunctionRepository/resources/CausalInvariantQ) are not included as well, because they should instead be properties of the [`WolframModelEvolutionObject`](https://github.com/maxitg/SetReplace/blob/master/README.md#properties), and have separate issues already.

#### Canonicalization/Isomorphism
- [ ] [`CanonicalWolframModelRule`](https://resources.wolframcloud.com/FunctionRepository/resources/CanonicalWolframModelRule)
- [ ] [`CanonicalHypergraph`](https://resources.wolframcloud.com/FunctionRepository/resources/CanonicalHypergraph)
- [ ] [`IsomorphicHypergraphQ`](https://resources.wolframcloud.com/FunctionRepository/resources/IsomorphicHypergraphQ)
- [ ] [`FindHypergraphIsomorphism`](https://resources.wolframcloud.com/FunctionRepository/resources/FindHypergraphIsomorphism) #77
  With a name like this, it should only find one possible isomorphism. If we want a function that finds all isomorphism (the format for which should definitely be different from the current WFR implementation), the name should be different.

#### Enumeration
For all functions above, I think we should also allow a single integer as the argument, specifying the complexity, `Length[Flatten[# /. Rule -> List]] &`(assuming atom names are not lists), which would then produce all rules/hypergraphs up to that complexity, including systems of multiple rules.

- [ ] [`EnumerateWolframModelRules`](https://resources.wolframcloud.com/FunctionRepository/resources/EnumerateWolframModelRules) #57
- [ ] [`RandomWoflramModel`](https://resources.wolframcloud.com/FunctionRepository/resources/RandomWolframModel)
- [ ] [`EnumerateHypergraphs`](https://resources.wolframcloud.com/FunctionRepository/resources/EnumerateHypergraphs)
- [ ] [`RandomHypergraph`](https://resources.wolframcloud.com/FunctionRepository/resources/RandomHypergraph)

#### Rule Properties
- [ ] [`ConnectedWolframModelRuleQ`](https://resources.wolframcloud.com/FunctionRepository/resources/ConnectedWolframModelQ)

#### Hypergraph Analysis Utilities
- [ ] [`ConnectedHypergraphQ`](https://resources.wolframcloud.com/FunctionRepository/resources/ConnectedHypergraphQ)
- [ ] [`HypergraphAdjacencyMatrix`](https://resources.wolframcloud.com/FunctionRepository/resources/HypergraphAdjacencyMatrix)
- [ ] `HypergraphIncidenceMatrix` would be useful as well.
- [ ] [`HypergraphNeighborhoods`](https://resources.wolframcloud.com/FunctionRepository/resources/HypergraphNeighborhoods) #413
  It should return hypergraphs (lists of lists), not graphs.
- [ ] [`HypergraphNeighborhoodVolumes`](https://resources.wolframcloud.com/FunctionRepository/resources/HypergraphNeighborhoodVolumes)
  I think we should define volume as the count of expressions, not atoms, so it should compute that instead. Maybe, we can have a second argument which would be either `"Expression"` or `"Atom"` depending on how we want to define the volume.
- [ ] [`HypergraphToGraph`](https://resources.wolframcloud.com/FunctionRepository/resources/HypergraphToGraph)
  There are multiple ways to do it. Perhaps a second argument would be useful to specify.
- [ ] [`GraphFunctionPlot`](https://resources.wolframcloud.com/FunctionRepository/resources/GraphFunctionPlot)
- [ ] `HypergraphFunctionPlot` should do the same, but for hypergraphs.

#### Graph/Hypergraph Construction
- [ ] [`HexagonalGridGraph`](https://resources.wolframcloud.com/FunctionRepository/resources/HexagonalGridGraph)
- [ ] [`HexagonalTorusGraph`](https://resources.wolframcloud.com/FunctionRepository/resources/HexagonalTorusGraph)
  Should be merged with the previous function, similar to [`GeneralizedGridGraph`](https://github.com/maxitg/SetReplace/#generalizedgridgraph).
- [ ] [`BuckyballGraph`](https://resources.wolframcloud.com/FunctionRepository/resources/BuckyballGraph)
- [ ] [`GeodesicSphereGraph`](https://resources.wolframcloud.com/FunctionRepository/resources/GeodesicSphereGraph)
- [ ] [`RegularHyperbolicTilingGraph`](https://resources.wolframcloud.com/FunctionRepository/resources/RegularHyperbolicTilingGraph)

#### Utilities
- [ ] [`ParallelMapMonitored`](https://resources.wolframcloud.com/FunctionRepository/resources/ParallelMapMonitored)
